crates/forge/tests/e2e/clean.rs

```diff
@@ -0,0 +1,241 @@
+        trace: false,
+    };
+
+    generate_clean_components(clean_components_state, &temp_dir);
+
+    runner(&temp_dir)
+        .arg("clean")
+        .arg("cache")
+        .assert()
+        .success();
+
+    let clean_components_state: CleanComponentsState = CleanComponentsState {
+        coverage: false,
+        profile: false,
+        cache: false,
+        trace: false,
+    };
+
+    assert_eq!(
+        check_clean_components_state(temp_dir.path()),
+        clean_components_state
+    );
+}
+
+#[test]
+#[cfg_attr(not(feature = "scarb_2_8_3"), ignore)]
+fn test_clean_all() {
+    let temp_dir = setup_package("coverage_project");
+
+    let clean_components_state = CleanComponentsState {
+        coverage: true,
+        cache: true,
+        trace: true,
+        profile: true,
+    };
+
+    generate_clean_components(clean_components_state, &temp_dir);
+
+    runner(&temp_dir).arg("clean").arg("all").assert().success();
+
+    let clean_components_state: CleanComponentsState = CleanComponentsState {
+        coverage: false,
+        cache: false,
+        trace: false,
+        profile: false,
+    };
+
+    assert_eq!(
+        check_clean_components_state(temp_dir.path()),
+        clean_components_state
+    );
+}
+
+#[test]
+fn test_clean_all_and_component() {
+    let temp_dir = setup_package("coverage_project");
+
+    let clean_components_state = CleanComponentsState {
+        coverage: false,
+        cache: true,
+        trace: true,
+        profile: false,
+    };
+    generate_clean_components(clean_components_state, &temp_dir);
+
+    // This command should fail because 'all' cannot be combined with other components
+    runner(&temp_dir)
+        .arg("clean")
+        .arg("all")
+        .arg("cache")
+        .assert()
+        .failure(); // "The 'all' component cannot be combined with other components"
+
+    let expected_state = CleanComponentsState {
+        coverage: false,
+        cache: true,
+        trace: true,
+        profile: false,
+    };
+
+    assert_eq!(
+        check_clean_components_state(temp_dir.path()),
+        expected_state
+    );
+}
+
+fn generate_clean_components(clean_components_state: CleanComponentsState, temp_dir: &TempDir) {
+    if clean_components_state.coverage {
+        assert!(clean_components_state.trace && clean_components_state.cache);
+        test_runner(temp_dir).arg("--coverage").assert().success();
+    }
+    if clean_components_state.profile {
+        assert!(clean_components_state.trace && clean_components_state.cache);
+        test_runner(temp_dir)
+            .arg("--build-profile")
+            .assert()
+            .success();
+    } else if clean_components_state.trace {
+        assert!(clean_components_state.cache);
+        test_runner(temp_dir)
+            .arg("--save-trace-data")
+            .assert()
+            .success();
+    } else if clean_components_state.cache {
+        test_runner(temp_dir).assert().success();
+    }
```

Comment:
We need to improve this logic - note that right now `test_runner` could be run more than once. Instead, let's collect flags (that depend on `clean_components_state`)  in `args` vector 😄 . Somethings like this

```rust
let args = ...
```

and later run it with:

```rust
runner.args(&args).assert().success();
```